http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/conn/IdleConnectionReaper.java

```diff
@@ -148,12 +148,13 @@ public void run() {
             while (!stopping) {
                 try {
                     Thread.sleep(sleepPeriod);
-
-                    for (Map.Entry<HttpClientConnectionManager, Long> entry : connectionManagers.entrySet()) {
-                        try {
-                            entry.getKey().closeIdleConnections(entry.getValue(), TimeUnit.MILLISECONDS);
-                        } catch (Exception t) {
-                            log.warn("Unable to close idle connections", t);
+                    if (!connectionManagers.isEmpty()) {
+                        for (Map.Entry<HttpClientConnectionManager, Long> entry : connectionManagers.entrySet()) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.



**Problem**  
 This line may not be thread-safe: **152**. At this line the following field is not protected by synchronization: `connectionManagers`. Your code typically protects this field with synchronization on object: `IdleConnectionReaper.this`.

**Fix**  
 Consider protecting the above field with synchronization on object: `IdleConnectionReaper.this`. For example, like at lines: **97** and **84**.

**More info**  
 The following lines are where the above field is already protected by synchronization on the above object: **84**, **97**, and **124**.